x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/create_rules_bulk_route.ts

```diff
@@ -51,13 +51,19 @@ export const createRulesBulkRoute = (
       logDeprecatedBulkEndpoint(logger, DETECTION_ENGINE_RULES_BULK_CREATE);
 
       const siemResponse = buildSiemResponse(response);
-      const rulesClient = context.alerting.getRulesClient();
-      const esClient = context.core.elasticsearch.client;
-      const savedObjectsClient = context.core.savedObjects.client;
-      const siemClient = context.securitySolution.getAppClient();
+
+      const core = await context.core;
+      const securitySolution = await context.securitySolution;
+      const licensing = await context.licensing;
+      const alerting = await context.alerting;
```

Comment:
Would it benefit from awaiting in parallel?

```ts
const [core, securitySolution, licensing, alerting] = Promise.all([context.core, context.securitySolution, context.licensing, context.alerting]);
```

Is it possible to provide a handy method for getting all the resolved parts that are needed?

```ts
const ctx = await context.resolve('core', 'securitySolution', 'licensing', 'alerting');
const esClient = ctx.core.elasticsearch.client;
```

Reply:
> Would it benefit from awaiting in parallel?

Not that much honestly given all context parts are awaiting their own dependencies, but that would be more elegant probably. 

> Is it possible to provide a handy method for getting all the resolved parts that are needed?

I really like the idea. Need to check the feasibility and impact given it will alter the `RequestHandlerContext` type.

Reply:
Done, and adapted the securitySolution handlers that will benefit the most from it in https://github.com/elastic/kibana/pull/129896/commits/a40bcdfa09fb9abefd7a03479cfb1742f437a5a5